### ISSUE TITLE
uAgents Support for Fetch Mainnet

### ISSUE THREAD

**Issue by harshpatel199259** [fetchai/uAgents#520]
### Prerequisites

- [X] I checked the [documentation](https://fetch.ai/docs) and made sure this feature does not already exist
- [X] I checked the [existing issues](https://github.com/fetchai/uAgents/issues) to make sure this feature has not already been requested

### Feature

Currently, uAgents register on the Dorado Testnet's Almanac contract, and their wallets are also on the Dorado Testnet. However, there is a need for uAgents to support the Fetch Mainnet, allowing users to hold and send real FET tokens (not testFET) through their uAgents.

**Comment by jrriehl** [comment]
While agent do not yet use of Fetch Mainnet as default, It should be possible to have your agent register on mainnet by setting the `test` flag to false:
```python
agent = Agent(..., test=False)
```

Let us know if you have any issues with this.

**Comment by harshpatel199259** [comment]
@jrriehl What I understand from your message is that by setting:
`agent = Agent(..., test=False)`

This will make my agent register on the Almanac smart contract using the mainnet, and the agent's wallet address will also be on the Fetch Mainnet. Does this mean that the Agent will need to pay with mainnet FET tokens for the registration process?

**Comment by Archento** [comment]
@harshpatel199259 Yes, exactly. The main net uses a different denomination and users will need to top up the wallet of the agent for it to be able to register. Currently that is FET but will be ASI in the near future.

But please be advised that we do not _explicitly_ recommend to use agents on the main net yet as that hasn't been tested thoroughly and since, e.g. we don't have enforcing mechanisms for the seed you need to be extra cautions about the uniqueness and security of that seed so that others are not able to access your agents funds.

**Comment by Autonomousweb3Dev** [comment]
> @harshpatel199259 Yes, exactly. The main net uses a different denomination and users will need to top up the wallet of the agent for it to be able to register. Currently that is FET but will be ASI in the near future.
> 
> But please be advised that we do not _explicitly_ recommend to use agents on the main net yet as that hasn't been tested thoroughly and since, e.g. we don't have enforcing mechanisms for the seed you need to be extra cautions about the uniqueness and security of that seed so that others are not able to access your agents funds.

I was wondering if there is any plan or ETA for thoroughly testing the uAgents on the Fetch Mainnet? It would be great to see an official announcement that enterprises and projects using Fetch agent tech can now confidently register uAgents on the mainnet. I believe prioritizing this will boost confidence among developers and could significantly increase adoption across projects.

Looking forward to hearing more about the roadmap and updates on this front!
